### PR TITLE
Add `Field` class

### LINKED ISSUE #9: Replace the types public API with the Field public API
In the reference implementation we moved from exposing types to exposing a [Field API](https://github.com/frictionlessdata/jsontableschema-py/blob/master/jsontableschema/field.py#L13). This was happening at the same time as the ruby port, so understandably is not here.

Also I think the change from `convert_*` to `cast_*` semantics occurred at around the same time. [Example](https://github.com/frictionlessdata/jsontableschema-py/blob/master/jsontableschema/field.py#L44).

it would be good to move the Ruby lib to the Field API.

```
# fields (new)
Field({'type': 'number'}).cast_value()

# types (old)
types.NumberType({'type': 'number'}).convert_value()
```

cc @roll

### PR DESCRIPTION
**Pull request by pezholio**
Fixes #9 

@roll, @pwalsh - Does this chime with your expectations in #9?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -209,14 +207,31 @@ rows = [
   ['wrong column count']
 ]
 
-schema.convert(rows)
+schema.cast(rows)
```
Comment (roll) [inline README.md]: From my experience almost all interactions with schema related to casting go thru `schema.cast_row` - not sure this method needed at all. Also name could be confusing (`schema.cast` - cast what? schema? data?). WDYT?

File: lib/jsontableschema.rb [hunk 2]
```diff
@@ -28,6 +28,7 @@
 require "jsontableschema/types/string"
```
Comment (roll) [inline lib/jsontableschema.rb]: I suppose it's something like `__init__.py` in Python? I'm not sure how it works in Ruby but in Python we use this to declare API - adding only public interface and not adding internal stuff (like types after field introduction). Like it's internal API could be changed anytime without any notification etc.

Comment (pezholio) [inline lib/jsontableschema.rb]: Yeah, that's it, this is the main file that requires everything needed by the gem. There are other patterns (i.e. we could modularise, so you could only require `jsonschema/table` or `jsonschema/schema`), but this is the generally accepted pattern, especially for small, focussed gems.

**Comment by pezholio** [comment]
@roll The schema and field connections are defined in https://github.com/theodi/jsontableschema.rb/blob/feature-add-field/lib/jsontableschema/model.rb - which is a Ruby module which is included in schema.rb - I've split a lot of things out into modules, which keeps things neater.

**Pull request merged**